File: benchmark.lua
-- Arithmetic operations
local arithSum = 0
for i = 0, 100000000 - 1 do
    arithSum = arithSum + i * 2
end

-- String concatenation
local strConcat = ""
for i = 0, 100000 - 1 do
    strConcat = strConcat .. "test"
end

-- Array operations
local arr = {}
for i = 0, 100000 - 1 do
    table.insert(arr, i)
end

local arrSum = 0
for i = 1, #arr do
    arrSum = arrSum + arr[i]
end

-- Function calls
local function square(n)
    return n * n
end

local funcSum = 0
for i = 0, 1000000 - 1 do
    funcSum = funcSum + square(i)
end

-- File write and read 1000 times
for j = 1, 1000 do
    local file = io.open("test.txt", "w")
    for i = 0, 10000 - 1 do
        file:write("This is a test file.\n")
    end
    file:close()

    file = io.open("test.txt", "r")
    local fileContent = file:read("*all")
    file:close()
end

-- Sorting
local unsortedArr = {}
for i = 0, 100000 - 1 do
    table.insert(unsortedArr, math.random())
end
table.sort(unsortedArr)

-- Dictionary operations (tables)
local dict = {}
for i = 0, 100000 - 1 do
    dict["key" .. i] = i
end

local dictSum = 0
for k, v in pairs(dict) do
    dictSum = dictSum + v
end

File: Benchmark.java
import java.io.*;
import java.util.*;

public class Benchmark {
  public static void main(String[] args) throws IOException {
    // Arithmetic operations
    long arithSum = 0;
    for (int i = 0; i < 100000000; i++) {
      arithSum += i * 2;
    }

    // String concatenation
    StringBuilder strConcat = new StringBuilder();
    for (int i = 0; i < 100000; i++) {
      strConcat.append("test");
    }

    // Array operations
    int[] arr = new int[100000];
    for (int i = 0; i < 100000; i++) {
      arr[i] = i;
    }
    long arrSum = 0;
    for (int num : arr) {
      arrSum += num;
    }

    // Function calls
    long funcSum = 0;
    for (int i = 0; i < 1000000; i++) {
      funcSum += square(i);
    }

    // File write and read 1000 times
    for (int j = 0; j < 1000; j++) {
      FileWriter writer = new FileWriter("test.txt");
      for (int i = 0; i < 10000; i++) {
        writer.write("This is a test file.\n");
      }
      writer.close();

      BufferedReader reader = new BufferedReader(new FileReader("test.txt"));
      StringBuilder fileContent = new StringBuilder();
      String line;
      while ((line = reader.readLine()) != null) {
        fileContent.append(line).append("\n");
      }
      reader.close();
    }

    // Sorting
    List<Double> unsortedArr = new ArrayList<>();
    for (int i = 0; i < 100000; i++) {
      unsortedArr.add(Math.random());
    }
    Collections.sort(unsortedArr);

    // HashMap operations
    HashMap<String, Integer> dict = new HashMap<>();
    for (int i = 0; i < 100000; i++) {
      dict.put("key" + i, i);
    }
    long dictSum = 0;
    for (Integer value : dict.values()) {
      dictSum += value;
    }
  }

  public static int square(int n) {
    return n * n;
  }
}
